Make the example documentation clearer (#105)
djc/quinn: README.md

Example certificates are included in the repository for test purposes.

```sh
$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./
$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
```

In the above example, the server will run on localhost and serve the "." folder to the client.
The client will request the "Cargo.toml" file.  If you wish to run across a network you need 
to update the `certs/openssl.cnf` file and change the `DNS.3` entry to suit the DNS name of 

> the server, and then regenerate the certificates using the `certs/generate.sh` script.

Author: I've updated the description with your previous suggestion.  I haven't tested the `dangerous_configuration` / `--accept-insecure-certs`, so am not really clear on how that works.

> the server, and then regenerate the certificates using the `certs/generate.sh` script.

Reviewer: `dangerous_configuration` is a cargo feature flag, which enables the rustls options that allow the default certificate verification procedure to be overriden, the quinn `ClientConfigBuilder::accept_insecure_certs` helper method to thereby disable verification entirely, and the corresponding command line flag in the client example, allowing a client to communicate with servers using invalid or untrusted (e.g. self-signed) certificates.

Example certificates are included in the repository for test purposes.

```sh
$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./
$ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml
```

In the above example, the server will run on localhost and serve the "." folder to the client.
The client will request the "Cargo.toml" file.  

> If you wish to run the client/server across a network for testing only (not production) you 

Reviewer: ```suggestion
To run the example client/server across a network you 
```
(and reflow)

need to update the `certs/openssl.cnf` file and change the `DNS.3` entry to suit the DNS name 
of the server, and then regenerate the certificates using the `certs/generate.sh` script.  This

> method of generating certificates is very limiting. The preferred method when running quinn 

Reviewer: ```suggestion
method of generating certificates is very limiting. The preferred method when running a 
```

> server on a network would be to use real certs from an actual certificate authority, like 

Reviewer: ```suggestion
server on a network is to use real certs from an actual certificate authority, like 
```

```sh
$ cd quinn
$ cargo run --example server -- --cert ./certs/server.chain --key ./certs/server.rsa ./

> $ cargo run --example client -- --ca ./certs/ca.der https://localhost:4433/Cargo.toml

Reviewer: I didn't realize cargo would search workspace members for examples! Please also remove the `cd quinn` line so this is consistent.

```

> The above example will work on localhost and serve the "Cargo.toml" file to the client.  If

Reviewer: This seems slightly misleading, as the server serves the entire directory supplied to it.

you wish to run across a network you need to update the `certs/openssl.cnf` file and change 

> the `DNS.3` entry to suit the DNS name of the server.

Reviewer: Wouldn't you need to regenerate the certificates after changing the configuration? Might also want to be clearer that you need to do this if you want to use these certs; the more typical way to run across a network would be to use real certs from e.g. Let's Encrypt.
